dataflow/src/main/java/org/checkerframework/dataflow/analysis/BackwardAnalysisImpl.java

```diff
@@ -0,0 +1,386 @@
+                            firstNode = node;
+                        }
+
+                        // propagate store to predecessors
+                        for (Block pred : rBlock.getPredecessors()) {
+                            propagateStoresTo(
+                                    pred,
+                                    firstNode,
+                                    currentInput,
+                                    FlowRule.EACH_TO_EACH,
+                                    addToWorklistAgain);
+                        }
+                        break;
+                    }
+
+                case EXCEPTION_BLOCK:
+                    {
+                        ExceptionBlock eBlock = (ExceptionBlock) block;
+
+                        TransferInput<V, S> inputAfter = getInput(eBlock);
+                        currentInput = inputAfter.copy();
+                        Node node = eBlock.getNode();
+                        TransferResult<V, S> transferResult =
+                                callTransferFunction(node, currentInput);
+                        boolean addToWorklistAgain = updateNodeValues(node, transferResult);
+
+                        // merged transferResult with exceptionStore if exist one
+                        S exceptionStore = exceptionStores.get(eBlock);
+                        S mergedStore =
+                                exceptionStore != null
+                                        ? transferResult
+                                                .getRegularStore()
+                                                .leastUpperBound(exceptionStore)
+                                        : transferResult.getRegularStore();
+
+                        for (Block pred : eBlock.getPredecessors()) {
+                            addStoreAfter(pred, node, mergedStore, addToWorklistAgain);
+                        }
+                        break;
+                    }
+
+                case CONDITIONAL_BLOCK:
+                    {
+                        ConditionalBlock cBlock = (ConditionalBlock) block;
+
+                        TransferInput<V, S> inputAfter = getInput(cBlock);
+                        TransferInput<V, S> input = inputAfter.copy();
+
+                        for (Block pred : cBlock.getPredecessors()) {
+                            propagateStoresTo(pred, null, input, FlowRule.EACH_TO_EACH, false);
+                        }
+                        break;
+                    }
+
+                case SPECIAL_BLOCK:
+                    {
+                        // special basic blocks are empty and cannot throw exceptions,
+                        // thus there is no need to perform any analysis.
+                        SpecialBlock sBlock = (SpecialBlock) block;
+                        final SpecialBlockType sType = sBlock.getSpecialType();
+                        // storage the store at entry
```

Comment:
store the storage? or just store at entry.
storage is a noun.

Reply:
I will look through the code and let's discuss again.